--- a/src/dxmt/dxmt_format.cpp
+++ b/src/dxmt/dxmt_format.cpp
@@ -778,12 +778,6 @@
   case DXGI_FORMAT_R16_FLOAT: {
     description.PixelFormat = WMTPixelFormatR16Float;
     description.AttributeFormat = WMTAttributeFormatHalf;
-    description.BytesPerTexel = 2;
-    break;
-  }
-  case DXGI_FORMAT_D16_UNORM: {
-    description.PixelFormat = WMTPixelFormatDepth16Unorm;
-    description.Flag = MTL_DXGI_FORMAT_DEPTH_PLANER;
     description.BytesPerTexel = 2;
     break;
   }
