--- a/src/d3d11/d3d11_device.cpp
+++ b/src/d3d11/d3d11_device.cpp
@@ -485,7 +485,7 @@
       if (Format == DXGI_FORMAT_R16G16B16A16_FLOAT || Format == DXGI_FORMAT_R10G10B10A2_UNORM)
         support1 &= ~D3D11_FORMAT_SUPPORT_DISPLAY;
     }
-    if (feature_level_ < D3D_FEATURE_LEVEL_10_1)
+    if (feature_level_ < D3D_FEATURE_LEVEL_9_1)
       support1 &= ~(D3D11_FORMAT_SUPPORT_SHADER_GATHER | D3D11_FORMAT_SUPPORT_SHADER_GATHER_COMPARISON);
     if (pFormatSupport && SUCCEEDED(hr))
       *pFormatSupport = support1;
