# recursive acquire returns the wrong HRESULT
diff --git a/src/dxmt/dxmt_keyed_mutex.cpp b/src/dxmt/dxmt_keyed_mutex.cpp
--- a/src/dxmt/dxmt_keyed_mutex.cpp
+++ b/src/dxmt/dxmt_keyed_mutex.cpp
@@ -125,5 +125,5 @@
 KeyedMutex::acquire(KeyedMutexAcquire &&func_acquire, uint64_t key, uint32_t milliseconds) {
   if (owned_.load(std::memory_order_acquire))
-    return DXGI_ERROR_INVALID_CALL;
+    return E_FAIL;

   LARGE_INTEGER timeout = {};
