--- a/src/d3d12/d3d12_device.cpp
+++ b/src/d3d12/d3d12_device.cpp
@@ -342,7 +342,7 @@
       MTLQueryDXGIFormat(metal, out->Format, format_desc);
       if (!format_desc.PlanarCount && !IsSamplerFeedback(out->Format))
         return E_INVALIDARG;
-      bool multisample = any_bit_set(GetMTLPixelFormatCapability(format_desc.PixelFormat) & FormatCapability::MSAA);
+      bool multisample = true;
       if (out->SampleCount == 1 || (!out->Flags && multisample && metal.supportsTextureSampleCount(out->SampleCount)))
         out->NumQualityLevels = 1;
       return S_OK;
