# each held buffer adds a separate swap chain reference
--- a/src/d3d12/d3d12_texture.cpp
+++ b/src/d3d12/d3d12_texture.cpp
@@ -933,7 +933,7 @@
   AddRef() override {
     std::lock_guard<dxmt::mutex> lock(buffers_.mutex);
     auto count = MTLD3D12Texture::AddRef();
-    if (count == 1 && !buffers_.referenced_buffers++)
+    if (count == 1 && ++buffers_.referenced_buffers)
       buffers_.swapchain->AddRef();
     return count;
   }
